react/LocationSelect.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import { useAddressContext } from 'vtex.address-context/AddressContext'
+import rules from './countries/rules'
+import { Dropdown } from 'vtex.styleguide'
+import { FormattedMessage, defineMessages } from 'react-intl'
+
+const messages = defineMessages({
+  province: {
+    defaultMessage: '',
+    id: 'place-components.label.province',
+  },
+  city: {
+    defaultMessage: '',
+    id: 'place-components.label.city',
+  },
+  department: {
+    defaultMessage: '',
+    id: 'place-components.label.department',
+  },
+  region: {
+    defaultMessage: '',
+    id: 'place-components.label.region',
+  },
+  community: {
+    defaultMessage: '',
+    id: 'place-components.label.community',
+  },
+  municipality: {
+    defaultMessage: '',
+    id: 'place-components.label.municipality',
+  },
+  district: {
+    defaultMessage: '',
+    id: 'place-components.label.district',
+  },
+  state: {
+    defaultMessage: '',
+    id: 'place-components.label.state',
+  },
+  locality: {
+    defaultMessage: '',
+    id: 'place-components.label.locality',
+  },
+})
+
+const LocationSelect: StorefrontFunctionComponent<{}> = () => {
+  const { address, setAddress } = useAddressContext()
+  const countryRules = rules[address.country]
+
+  if (!countryRules.locationSelect)
+    throw 'The LocationSelect component is not applicable to this country :('
+
+  const { countryData, fields } = countryRules.locationSelect
+
+  const getCompletedFieldsArray = () => {
+    let completedFields = []
+    for (let field of fields) {
+      if (address[field.name]) {
+        completedFields.push(address[field.name])
+      } else {
+        break
+      }
+    }
+    return completedFields
+  }
```

Comment:
Can't you rewrite this as a `map-filter`? I tend to prefer functional style because it describes more clearly what you're doing.

Reply:
I don't know how to do this with `map-filter` directly, since the filtering would be kind of strange (I need to stop taking elements after I find the first `null` element).
I believe the functional way to do this would be with a `takeWhile` function, this is the best I could get;
```typescript
const takeWhile = (array: any[], f: (arg0: any) => boolean) => {
  const first = array.findIndex(f)
  return first === -1 ? array : array.slice(0, first)
}
const addressFields = fields.map(field => address[field.name])
return takeWhile(addressFields, x => !x)
```
What do you think about this?

Reply:
Sorry, I completely missed the `break` in the code, so my initial idea would not work.
Still, I think I like this style more, but I'd like some second opinion on this. I would just remove these `getCompletedFieldsArray` and `takeWhile` functions, which you use only once, and write simply
```ts
const addressFields = fields.map(field => address[field.name])
const firstMissingIdx = addressFields.findIndex(field => !field)
const completedFields = firstMissingIdx === -1 
  ? addressFields 
  : addressFields.slice(0, firstMissngIdx)
```

Reply:
That seems awesome, thanks!